protocol/avdl/keybase1/gui_file_access.avdl

```diff
@@ -0,0 +1,129 @@
+@namespace("keybase.1")
+
+protocol GuiFileAccess {
+
+  import idl "common.avdl";
+
+/*
+  File manipulation interface
+  
+  Caller provides OpID to asynchronous method call, repeats until status is no
+  longer pending or operation is cancelled.
+ 
+  KBFS paths are expected to start with "private" or "public" - mount name is implied.
+*/
+
+  // Random GUID provided by caller 
+  fixed OpID(16);
+  
+  enum ItemType {
```

Comment:
You might need `SYM` as well for symlinks, depending on whether the GUI wants to treat those differently from other files.  If not, the service will need to resolve those to figure out the true type.

We also have an `EXEC` type, if the GUI wants to know executability on non-windows platforms.